src/views/open-api-docs-view/components/open-api-sidebar-contents/index.tsx

```diff
@@ -3,42 +3,87 @@
  * SPDX-License-Identifier: MPL-2.0
  */
 
-import { OperationGroup, OperationProps } from 'views/open-api-docs-view/types'
+import { useMemo } from 'react'
+import { useRouter } from 'next/router'
+// Lib
+import { useActiveSection } from 'lib/hash-links/use-active-section'
+// Components
+import { SidebarNavMenuItem } from 'components/sidebar/components'
+// Types
+import type { OpenApiNavItem } from 'views/open-api-docs-view/types'
+// Styles
+import s from './open-api-sidebar-contents.module.css'
 
 /**
  * Renders sidebar contents for OpenApiDocsView.
- *
- * TODO: implement this component.
- * Designs show a list of navigational items
- * that link to the corresponding #operation-slug on the page.
- * We also want to highlight the current navigational item.
- *
- * Note: we likely also want to re-use this component to render
- * the OpenApiDocsView level of the corresponding mobile nav.
  */
 export function OpenApiSidebarContents({
-	operationGroups,
+	navItems,
 }: {
-	operationGroups: OperationGroup[]
+	navItems: OpenApiNavItem[]
 }) {
+	const { asPath } = useRouter()
+
+	/**
+	 * Note next/router `asPath` returns *with* the `#hash`.
+	 *
+	 * Here, we remove the hash, so that we can compare to the `fullPath`
+	 * of each nav item, which are not expected to have hashes,
+	 * to the URL path that we're on.
+	 *
+	 * Note that domain does not matter, as we're just grabbing the hash
+	 */
+	const urlPathname = useMemo(
+		() => new URL(asPath, 'https://www.example.com').pathname,
+		[asPath]
+	)
+
+	/**
+	 * Build an array of section slugs,
+	 * and pass these to `useActiveSection`.
+	 */
+	const sectionSlugs = useMemo(() => {
+		return (
+			navItems
+				.map((item) => {
+					// Only grab slugs from link items
+					if (!('fullPath' in item)) {
+						return null
+					}
+					// Note that domain does not matter, we're just grabbing the hash
+					const { hash } = new URL(item.fullPath, 'https://www.example.com')
```

Comment:
💡 Could be room to create a shared lib function to create the URL here with the 'stub' domain. I do find that we've used this type of thing quite a bit in the codebase lately.

Reply:
I dig it! Made a very smol wrapper around `new URL(base, domain)` that just makes `domain` optional, giving `https://www.example.com` as the default 👍